lib/ethpy/ethpy/base/transactions.py

```diff
@@ -390,15 +417,11 @@ async def async_smart_contract_transact(
             function_name_or_signature=function_name_or_signature,
             fn_args=fn_args,
             fn_kwargs={},
+            raw_txn=dict(unsent_txn),
         ) from err
 
 
-def _send_transaction_and_wait_for_receipt(
-    func_handle: ContractFunction,
-    signer: LocalAccount,
-    web3: Web3,
-    nonce: Nonce | None = None,
-) -> TxReceipt:
+def _send_transaction_and_wait_for_receipt(unsent_txn: TxParams, signer: LocalAccount, web3: Web3) -> TxReceipt:
```

Comment:
how are we handling nonce now?

also nonce is still in the docstring for this func

Reply:
Nonce is getting set in build transaction